includes/wc-formatting-functions.php

```diff
@@ -1117,7 +1117,8 @@ function wc_format_option_hold_stock_minutes( $value, $option, $raw_value ) {
 	wp_clear_scheduled_hook( 'woocommerce_cancel_unpaid_orders' );
 
 	if ( '' !== $value ) {
-		wp_schedule_single_event( time() + ( absint( $value ) * 60 ), 'woocommerce_cancel_unpaid_orders' );
+		$duration = apply_filters( 'woocommerce_cancel_unpaid_orders_duration', $value );
```

Comment:
The problem with this change is that this function fires when someone updates the Hold Stock option in the settings, having a filter here might be confusing when a user enters 100 in the settings and saves and the cron is then scheduled for 20minutes instead because it was changed via the filter here. A better solution to this might be to hide the input box for the setting when the filter is in use and just display the value on the setting page.

Reply:
If this can be set via the settings input box, what would be the real value of adding a filter to it?